gnomad/utils/constraint.py

```diff
@@ -877,3 +839,344 @@ def compute_expected_variants(
     agg_expr.update({ann: agg_func(ht[ann]) for ann in ann_to_sum})
 
     return agg_expr
+
+
+def oe_aggregation_expr(
+    ht: hl.Table,
+    filter_expr: hl.expr.BooleanExpression,
+    pops: Tuple[str] = (),
+    exclude_mu_sum: bool = False,
+) -> hl.expr.StructExpression:
+    """
+    Get aggregation expressions to compute the observed:expected ratio for rows defined by `filter_expr`.
+
+    Return a Struct containing aggregation expressions to sum the number of observed
+    variants, possible variants, expected variants, and mutation rate (if
+    `exclude_mu_sum` is not True) for rows defined by `filter_expr`. The Struct also
+    includes an aggregation expression for the observed:expected ratio.
+
+    The following annotations are in the returned StructExpression:
+        - obs - the sum of observed variants filtered to `filter_expr`.
+        - mu - the sum of mutation rate of variants filtered to `filter_expr`.
+        - possible - possible number of variants filtered to `filter_expr`.
+        - exp - expected number of variants filtered to `filter_expr`.
+        - oe - observed:expected ratio of variants filtered to `filter_expr`.
+
+        If `pops` is specified:
+            - pop_exp - Struct with the expected number of variants per population (for
+              all pop in `pops`) filtered to `filter_expr`.
+            - pop_obs - Struct with the observed number of variants per population (for
+              all pop in `pops`) filtered to `filter_expr`.
+
+
+    .. note::
+        The following annotations should be present in `ht`:
+            - observed_variants
+            - mu
+            - possible_variants
+            - expected_variants
+        If `pops` is specified, the following annotations should also be present:
+            - expected_variants_{pop} for all pop in `pops`
+            - downsampling_counts_{pop} for all pop in `pops`
+
+    :param ht: Input Table to create observed:expected ratio aggregation expressions for.
+    :param filter_expr: Boolean expression used to filter `ht` before aggregation.
+    :param pops: List of populations to compute constraint metrics for. Default is ().
+    :param exclude_mu_sum: Whether to exclude mu sum aggregation expression from
+        returned struct. Default is False.
+    :return: StructExpression with observed:expected ratio aggregation expressions.
+    """
+    # Create aggregators that sum the number of observed variants, possible variants,
+    # and expected variants and compute observed:expected ratio.
+    agg_expr = {
+        "obs": hl.agg.sum(ht.observed_variants),
+        "exp": hl.agg.sum(ht.expected_variants),
+        "possible": hl.agg.sum(ht.possible_variants),
+    }
+    agg_expr["oe"] = agg_expr["obs"] / agg_expr["exp"]
```

Comment:
I changed this in the PR to use `divide_null` to handle cases where `agg_expr["exp"]` is 0